src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -409,16 +508,91 @@ private void ProcessProjectChange(Solution solution, ProjectId projectId)
+            {
+                var parentHierarchy = TryGetXProjParentHierarchy(documentId, hierarchy, hostProject);
+                if (parentHierarchy != null)
+                {
+                    Guid projectGuid;
+                    if (ErrorHandler.Succeeded(parentHierarchy.GetGuidProperty(VSConstants.VSITEMID_ROOT, (int)__VSHPROPID.VSHPROPID_TypeGuid, out projectGuid)) &&
+                        projectGuid == XProjProjectGuid)
+                    {
+                        // We have an xproj project.  Package installation is disabled here until 
+                        // they let us know it's ok for us to take over that role/
+                        //
+                        // We do this through a special "hand off" property they expose on their 
+                        // hierarchy. If that property is present, then xproj is "handing off" 
+                        // responsibility for this feature to us.  In that case we are allowed to 
+                        // to light up the "install package" feature for xproj projects.  Otherwise,
+                        // XProj if the "hand off" is not present, then xproj owns this experience.
+                        //
+                        // Note: this is only necessary for the transition period between shipping
+                        // VS2015 Update 2 and the next XProj release.
+                        const int HandOffPropertyId = 870022618;
+                        object handOffProperty;
+                        return parentHierarchy.TryGetProperty(HandOffPropertyId, out handOffProperty) &&
+                            handOffProperty != null;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        private static IVsHierarchy TryGetXProjParentHierarchy(DocumentId documentId, IVsHierarchy hierarchy, IVisualStudioHostProject hostProject)
+        {
+            // In the xproj case we'll have their documents parented by a hierarchy per target 
+            // type.  i.e. a hierarchy for when they're targetting DNX 4.5.1 and a hierarchy 
+            // when they're targetting DNS Core 5.  These hierarchies will be parented by another
+            // actual hierachy.  This hierarchy will have a Guid on it that indicates that it is
+            // xproj.  We can try to get this parent hierarchy by asking for the "Shared Project
+            // Hierarchy" for the document we're currently examining.
+            var hostDocument = hostProject.GetDocumentOrAdditionalDocument(documentId);
+            if (hostDocument != null)
+            {
+                var hostDocumentItemId = hostDocument.GetItemId();
+
+                try
+                {
+                    object pvarParentHierarchy;
+                    if (ErrorHandler.Succeeded(hierarchy.GetProperty(
+                        hostDocumentItemId, (int)__VSHPROPID7.VSHPROPID_SharedProjectHierarchy, out pvarParentHierarchy)))
```

Comment:
@dpoeschl - didn't we want to stop using this property in the case of xproj and use something else specific to xproj instead, or was that for going in the reverse direction?